sympy/core/expr.py

```diff
@@ -813,11 +813,11 @@ def _eval_interval(self, x, a, b):
             return B - A
 
         value = B - A
-        if all([a.is_comparable, b.is_comparable]):
+        if a.is_comparable and b.is_comparable:
             singularities = list(solveset(self.cancel().as_numer_denom()[1], x))
             for s in singularities:
-                if a < s < b:
-                    value += -limit(self, x, s, "+") + limit(self, x, s, "-")
+                    if a < s < b or b < s < a:
```

Comment:
Anyways, `_eval_interval` is an internal method and I think it always expects `b >= a`.The current code returns `B - A` which roughly corresponds to `F(b) - F(a)`.